dds_web/api/project.py

```diff
@@ -183,20 +183,69 @@ def post(self):
         ]:
             raise DDSArgumentError("Invalid status")
 
+        curr_date = dds_web.utils.current_time()
+        add_aborted = {}
+        add_deadline = {}
+
         if not self.is_transition_possible(project.current_status, new_status):
             raise DDSArgumentError("Invalid status transition")
 
-        try:
-            add_status = models.ProjectStatuses(
-                **{
-                    "project_id": project.id,
-                    "status": new_status,
-                    "date_created": dds_web.utils.current_time(),
-                }
-            )
+        # Moving to Available
+        if new_status == "Available":
+            # Optional int arg deadline in days
+            deadline = extra_args.get("deadline", project.responsible_unit.days_to_expire)
+            if project.current_status == "Expired":
+                # Project can only move from Expired 2 times
+                if project.times_expired > 2:
+                    raise DDSArgumentError(
+                        "Project availability limit: Project cannot be made Available any more times"
+                    )
 
+                add_deadline["deadline"] = dds_web.utils.current_time(
+                    to_midnight=True
+                ) + datetime.timedelta(days=deadline)
+
+            if not project.has_been_available:
+                project.released = curr_date
+                add_deadline["deadline"] = dds_web.utils.current_time(
+                    to_midnight=True
+                ) + datetime.timedelta(days=deadline)
+
+        # Moving to Expired
+        if new_status == "Expired":
```

Comment:
This should be an automatic thing, probably not changable by the actual users. But we can have this here for now anyway, until we create the scheduler.

Reply:
I agree that it should be an automatic thing in the usual course but it'll be nice to have a manual option just in case.

Reply:
Ok! I'm not sure I see why unit admins / personnel should be able to do this though, so in that case I'd limit it to `Super Admins`